### PR TITLE
Add support for delta encoding of attributes in ReportRequest

### PR DESCRIPTION
**Pull request by geeknoid**
Also, stop returning APA-produced attributes with every request.

This bumps code coverage of pkg/api to 100%.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/790)

### CODE REVIEW

File: pkg/attribute/mutableBag.go [hunk 1]
```diff
@@ -305,73 +319,79 @@ func GetBagFromProto(attrs *mixerpb.Attributes, globalDict []string) (*MutableBa
 	// TODO: fail if the proto carries multiple attributes by the same name (but different types)
 
 	var buf *bytes.Buffer
+	log := func(format string, args ...interface{}) {}
```
Comment (ZackButcher) [inline pkg/attribute/mutableBag.go]: Nice, much cleaner.

File: pkg/api/grpcServer.go [hunk 2]
```diff
@@ -87,33 +96,27 @@ func (s *grpcServer) dispatch(ctx context.Context, dState *dispatchState, worker
 		return makeGRPCError(out)
 	}
 
-	responseBag := attribute.GetMutableBag(nil)
+	preprocResponseBag := attribute.GetMutableBag(requestBag)
```
Comment (douglas-reid) [inline pkg/api/grpcServer.go]: does this get serialized anywhere? it'd be nice to dump this all in one place, perhaps close to where we log the response (for debug levels).

File: pkg/attribute/mutableBag.go [hunk 3]
```diff
@@ -305,73 +319,79 @@ func GetBagFromProto(attrs *mixerpb.Attributes, globalDict []string) (*MutableBa
...
+
 	if glog.V(2) {
 		buf = pool.GetBuffer()
-		log(buf, "Creating bag from wire attributes:\n")
+		log = func(format string, args ...interface{}) {
+			fmt.Fprintf(buf, format, args...)
```
Comment (douglas-reid) [inline pkg/attribute/mutableBag.go]: suggestion: all the log() calls end with "\n". perhaps add that detail here, and save having to add it to each call.

maybe something like:
```
if buf.Bytes()[buf.Len()-1] != '\n' {
  buf.WriteByte('\n')
}
```

File: pkg/api/grpcServer.go (hunk 2 above)
Comment (geeknoid) [inline pkg/api/grpcServer.go]: I added code to dump out all the attributes before invoking the normal adapters.

File: pkg/attribute/mutableBag.go (hunk 3 above)
Comment (geeknoid) [inline pkg/attribute/mutableBag.go]: Done.

**Pull request merged**